jujugui/static/gui/src/app/app.js

```diff
@@ -801,7 +801,8 @@ YUI.add('juju-gui', function(Y) {
           instanceName="white-box"
           visible={visible}>
           <components.SearchResults
-            query={text} />
+            query={text}
+            charmstore={this.get('charmstore')} />
```

Comment:
We've been using a uni-directional data flow pattern, so components don't interact with a model, only call supplied methods. See how we've done this here: https://github.com/juju/juju-gui/blob/pyramid-fork/jujugui/static/gui/src/app/app.js#L813